crates/ruff/src/directives.rs

```diff
@@ -95,6 +95,27 @@ fn extract_noqa_line_for(lxr: &[LexResult], locator: &Locator, indexer: &Indexer
                 break;
             }
 
+            // F-strings can be multi-line even without triple quotes. For example,
+            //
+            // ```python
+            // f"foo {
+            //    x
+            //        *
+            //             y
+            // }"
+            // ```
+            //
+            // We expect `noqa` directives on the last line of the f-string.
```

Comment:
I would expect the noqa directive to be added on the line of `x` if x has a violation

Reply:
But then it doesn't align with how it gets added with triple-quoted f-strings. In that the directive gets added on the last line of the string:

```python
f"""test{
    a
        *
            b
}test"""  # noqa: F821
```

Reply:
I would also expect this noqa to be added to the expression because it is a multiline string. 

I think there are multiple questions here that we haven't talked about

1. How should noqa comments in fstrings work in Python 3.12 forward, now that comments are allowed inside f-strings
2. Whether we want to implement the new layout now or defer to later
3. How to implement the change